dot/network/service.go

```diff
@@ -72,6 +74,9 @@ type Service struct {
 	mdns      *mdns
 	gossip    *gossip
 	syncQueue *syncQueue
+	bufPool   *sync.Pool
+	bufMap    *sync.Map //map[*[maxMessageSize]byte]struct{}
+	bufCount  int
```

Comment:
You can also use channels instead of `bufMap` for the keeping the reference of the buffer.
Using channel you can easily get it's length so need of `bufCount`

Reply:
removed in favour of `sizedBufferPool`